app/models/user.rb

```diff
@@ -13,4 +13,17 @@ class User< ActiveRecord::Base
   # end
   # ucheckout
   # end
+
+  def self.from_omniauth(data)
+    user = User.where(:email => data["info"]["email"]).first
+    unless user
+      user = User.create(
+        name:data["info"]["name"],
+        email: data["info"]["email"],
+        password: [*('a'..'z'),*('0'..'9')].sample(8).join
```

Comment:
Space missing after comma.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.